### DISCUSSION TITLE
CSLA 6.2 and LinqToSql

### DISCUSSION THREAD

**Discussion by DMFullbrook** [MarimerLLC/csla#3181] [Questions]
Upgrading to CSLA 6.2. I'm beginning to think this is a non-Starter. My project uses LinqToSql throughout. Here is an example of a simple Fetch in the DataAccess.LinqToSql layer:

```
        public CompanyDto Fetch(Guid federationCompanyId)
        {
            CompanyDto result = default(CompanyDto);

            using (var ctx = ContextManager<FederationDataContext>.GetManager(this._connectionConfiguration.ConnectionName))
            {
                ctx.DataContext.CommandTimeout = this._connectionConfiguration.CommandTimeout;
                result = ctx.DataContext.stp_Company_GetCompanyById(federationCompanyId).SingleOrDefault();
            }

            return result;
        }
```

Beautifully simple but I understand that LinqToSql has been deprecated in the ContextManager. Is that correct? What's the alternative? All help gratefully appreciated.

**Comment by rockfordlhotka** [comment]
Things like the `ContextManager` have been replaced by the use of dependency injection.

Have you read this document?

https://github.com/MarimerLLC/csla/blob/main/docs/Upgrading%20to%20CSLA%206.md

**Comment by TheCakeMonster** [comment]
Entity framework is probably the obvious replacement for anything using LinqToSql. That doesn't mean that it has to be replaced. However, EF is something for which there is lots of help and support and which is much more common in the industry now.

If you want to keep your LinqToSql implementation then you would have to look at injecting something like the context manager. You can have CSLA inject properties into your Fetch method by adorning them with the [Inject] attribute. Remember that you also have to register something into DI for that to work.

**Reply by TheCakeMonster** [reply]
The Blazor example contains a data access example that uses EF. It uses the repository pattern to abstract the data access to allow for different implementations, including a mock data access layer for testing. You don't have to use the repository pattern, but it does bring the advantages of easier unit testing.

https://github.com/MarimerLLC/csla/blob/main/Samples/BlazorExample/BlazorExample/DataAccess.EF/PersonEFDal.cs

**Reply by DMFullbrook** [reply]
Thanks. I guess my follow-on question to your suggestion to inject a ContextManager is where can this be found? Since the CSLA ContextManager no longer supports LinqToSql?

**Reply by rockfordlhotka** [reply]
In "modern" DI-based code you'd inject the database connection object, not a connection manager. The DI infrastructure will ensure that the connection object is disposed at the end when the DI scope is disposed.
